stdlib/public/core/Sort.swift

```diff
@@ -516,35 +516,34 @@ internal func _siftDown<C: MutableCollection & RandomAccessCollection>(
   subRange range: Range<C.Index>,
   by areInIncreasingOrder: (C.Element, C.Element) throws -> Bool
 ) rethrows {
-
-  let countToIndex = elements.distance(from: range.lowerBound, to: index)
-  let countFromIndex = elements.distance(from: index, to: range.upperBound)
-  // Check if left child is within bounds. If not, return, because there are
+  var index = index
+  var countToIndex = elements.distance(from: range.lowerBound, to: index)
+  var countFromIndex = elements.distance(from: index, to: range.upperBound)
+  // Check if left child is within bounds. If not, stop iterating, because there are
   // no children of the given node in the heap.
-  if countToIndex + 1 >= countFromIndex {
-    return
-  }
-  let left = elements.index(index, offsetBy: countToIndex + 1)
-  var largest = index
-  if (try areInIncreasingOrder(elements[largest], elements[left])) {
-    largest = left
-  }
-  // Check if right child is also within bounds before trying to examine it.
-  if countToIndex + 2 < countFromIndex {
-    let right = elements.index(after: left)
-    if (try areInIncreasingOrder(elements[largest], elements[right])) {
-      largest = right
+  while countToIndex + 1 < countFromIndex {
+    let left = elements.index(index, offsetBy: countToIndex + 1)
+    var largest = index
+    if (try areInIncreasingOrder(elements[largest], elements[left])) {
```

Comment:
tiny nit, and I realize you're imitating the style in the rest of the file (which I think predates 1.0): you don't have to put parentheses around conditions in an `if`.